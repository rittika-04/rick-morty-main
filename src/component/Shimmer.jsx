

const Shimmer = () =>{

    return(
      <div className="shimmer-container" data-testid = "shimmer-container">
        {Array(20).fill("").map((_,index) =>(
            <div key = {index} className="shimmer-card" data-testid = "shimmer-card">
            </div>
        ))}
      </div>
    )
}

export default Shimmer;