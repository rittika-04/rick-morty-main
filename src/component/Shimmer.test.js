
import { render,screen } from "@testing-library/react";
import Shimmer from "./Shimmer";

describe('Shimmer',()=>{
    it("all went good, we tested 20 card", ()=>{
        render(<Shimmer/>);
        
        // we use screen for finding element
        const shimercard = screen.getAllByTestId('shimmer-card');
        expect(shimercard.length).toBe(20);
    });

    it("correcly render component",()=>{
        render(<Shimmer/>)

        const shimmerrendering = screen.getByTestId('shimmer-container');
        expect(shimmerrendering).toBeInTheDocument();
    })

    it("matches the snapshot", ()=>{

        const {asFragment} =  render(<Shimmer/>);
        expect(asFragment()).toMatchSnapshot();
    })
});


