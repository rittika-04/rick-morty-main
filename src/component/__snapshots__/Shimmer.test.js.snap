// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shimmer matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="shimmer-container"
    data-testid="shimmer-container"
  >
    <div
      class="shimmer-card"
      data-testid="shimmer-card"
    />
    <div
      class="shimmer-card"
      data-testid="shimmer-card"
    />
    <div
      class="shimmer-card"
      data-testid="shimmer-card"
    />
    <div
      class="shimmer-card"
      data-testid="shimmer-card"
    />
    <div
      class="shimmer-card"
      data-testid="shimmer-card"
    />
    <div
      class="shimmer-card"
      data-testid="shimmer-card"
    />
    <div
      class="shimmer-card"
      data-testid="shimmer-card"
    />
    <div
      class="shimmer-card"
      data-testid="shimmer-card"
    />
    <div
      class="shimmer-card"
      data-testid="shimmer-card"
    />
    <div
      class="shimmer-card"
      data-testid="shimmer-card"
    />
    <div
      class="shimmer-card"
      data-testid="shimmer-card"
    />
    <div
      class="shimmer-card"
      data-testid="shimmer-card"
    />
    <div
      class="shimmer-card"
      data-testid="shimmer-card"
    />
    <div
      class="shimmer-card"
      data-testid="shimmer-card"
    />
    <div
      class="shimmer-card"
      data-testid="shimmer-card"
    />
    <div
      class="shimmer-card"
      data-testid="shimmer-card"
    />
    <div
      class="shimmer-card"
      data-testid="shimmer-card"
    />
    <div
      class="shimmer-card"
      data-testid="shimmer-card"
    />
    <div
      class="shimmer-card"
      data-testid="shimmer-card"
    />
    <div
      class="shimmer-card"
      data-testid="shimmer-card"
    />
  </div>
</DocumentFragment>
`;
